### DISCUSSION TITLE
Immer Middelware Typescript error: Exported variable 'useMyStore' has or is using name 'WritableDraft' from external module "../node_modules/immer/dist/immer" but cannot be named.

### DISCUSSION THREAD

**Discussion by rohitspujari** [pmndrs/zustand#2429] [Q&A]
When I add an array property of a custom type like `Cites: City[]`, the build fails with the error above. But if I use `City: any[]`, or a singular property like string or even a singular custom type `city: City`, it works fine. 

Here is how I configured my store 

`export const useMyStore = create<MyStore>()(
  immer((set, get) => ({
   ...
 }))
`
I'm using Vite for the build.

**Comment by dbritto-dev** [comment]
@rohitspujari would you mind sharing us a minimal repro on stackblitz?

**Comment by jessekrubin** [comment]
@dbritto-dev (and @rohitspujari ?) here is a repro... I get the same issue... https://github.com/jessekrubin/zustand-middleware-immer-writeable-draft-tsc

**Reply by jessekrubin** [reply]
`pnpm install && ./node_modules/.bin/tsc`

**Reply by dbritto-dev** [reply]
Thanks I'll look into that

**Reply by dai-shi** [reply]
With the latest versions, I don't get errors.
https://stackblitz.com/github/jessekrubin/zustand-middleware-immer-writeable-draft-tsc?file=package.json,src%2Flib.ts

**Reply by dai-shi** [reply]
Wait, I might misunderstood something.

**Reply by jessekrubin** [reply]
(@dai-shi while youre here in this thread, GREATE WORK ON THIS AND EVERYTHING YOU HAVE MADE! I use zustand and jotai extensively!)

**Comment by walkjoi** [comment]
same problem here. 

And I find the reproduction steps:
1. Define a state variable in the store.
2. Assign a nested type definition to the state variable.
3. Observe the error message mentioned above.

<img width="1287" alt="image" src="https://github.com/pmndrs/zustand/assets/32947565/8f5917da-c780-422f-941b-255f9d5c4010">

**Reply by walkjoi** [reply]
"immer": "10.0.3",
    "zustand": "4.5.1"

**Reply by dai-shi** [reply]
Can you try the latest versions?
```js
    "immer": "^10.1.1",
    "react": "^18.3.1",
    "zustand": "^4.5.2"
```

**Reply by dai-shi** [reply]
Wait, I might misunderstood something.

Can you create a reproduction with stackblitz?

**Reply by walkjoi** [reply]
I'm not familiar with stackblitz, hope this will help: https://stackblitz.com/edit/vitejs-vite-uuqfzh?file=src%2Fstore.ts

**Reply by dai-shi** [reply]
Thanks. Here's a modified version: https://stackblitz.com/edit/vitejs-vite-xbxksa?file=src%2Fstore.ts,package.json

It seems working. No errors. I wonder how the error can be reproduced.
